mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.regression
+
+import org.apache.spark.api.java.{JavaRDD, JavaPairRDD}
+import org.apache.spark.rdd.RDD
+
+/**
+ * Regression model for Isotonic regression
+ *
+ * @param predictions Weights computed for every feature.
```

Comment:
There are three Doubles in the input. It is necessary to explain what they are. Essentially, this is a lookup table. To make it efficient, we should store the ordered bounds in `Array[Double]` and the mapped values in another `Array[Double]`. For example,

~~~
[-inf, 1.0) -> [0.5]
[1.0, 2.0) -> [1.5]
[3.0, 4.0) -> [3.5]
[4.0, inf) -> [4.5]
~~~

can be stored as [1.0, 2.0, 3.0, 4.0] and [0.5, 1.5, 3.5, 4.5]. Then you can use binary search to find the insert index and then the mapped value quickly.